ofp4/tests/test.rs

```diff
@@ -0,0 +1,444 @@
+        .collect();
+
+    let (_cleanup, tmp_dir, client) = start_ofp4(p4info).await?;
+
+    // Add a multicast group entry, with ID 1, that contains ports 1, 2, 3, and 4.
+    fn replica(egress_port: u32, instance: u32) -> Replica {
+        Replica { egress_port, instance, ..Default::default() }
+    }
+    let mge = MulticastGroupEntry {
+        multicast_group_id: 1,
+        replicas: RepeatedField::from_vec(vec![
+            replica(1, 1),
+            replica(2, 1),
+            replica(3, 1),
+            replica(4, 1),
+        ]),
+        ..Default::default()
+    };
+    let pree = PacketReplicationEngineEntry {
+        field_type: Some(PacketReplicationEngineEntry_oneof_type::multicast_group_entry(mge)),
+        ..Default::default()
+    };
+    let entity = Entity {
+        entity: Some(Entity_oneof_entity::packet_replication_engine_entry(pree)).into(),
+        ..Default::default()
+    };
+    let mge_update = Update {
+        field_type: Update_Type::INSERT,
+        entity: Some(entity).into(),
+        ..Default::default()
+    };
+
+    // Add tagged VLAN with ID 1.
+    let table = &tables["SnvsIngress.InputVlan"];
+    let mfs = &table.match_fields;
+    fn exact_fm(mf: &MatchField, value: Vec<u8>) -> FieldMatch {
+        let exact = FieldMatch_Exact { value, ..Default::default() };
+        FieldMatch {
+            field_id: mf.preamble.id,
+            field_match_type: Some(FieldMatch_oneof_field_match_type::exact(exact)).into(),
+            ..Default::default()
+        }
+    }
+    let fms = vec![exact_fm(&mfs[0], vec![0, 1]),
+                   exact_fm(&mfs[1], vec![1])];
+    let action = Action {
+        action_id: actions["SnvsIngress.UseTaggedVlan"],
+        ..Default::default()
+    };
+    let table_action = TableAction {
+        field_type: Some(TableAction_oneof_type::action(action)),
+        ..Default::default()
+    };
+    let te = TableEntry {
+        table_id: table.preamble.id,
+        field_match: RepeatedField::from_vec(fms),
+        action: Some(table_action).into(),
+        priority: 50,
+        ..Default::default()
+    };
+    let entity = Entity {
+        entity: Some(Entity_oneof_entity::table_entry(te)).into(),
+        ..Default::default()
+    };
+    let te_update = Update {
+        field_type: Update_Type::INSERT,
+        entity: Some(entity).into(),
+        ..Default::default()
+    };
+    let updates = vec![te_update, mge_update];
+    let wr = WriteRequest {
+        device_id: DEVICE_ID,
+        election_id: Some(election_id()).into(),
+        updates: RepeatedField::from_vec(updates),
+        atomicity: WriteRequest_Atomicity::DATAPLANE_ATOMIC,
+        ..Default::default()
+    };
+    client.write(&wr)?;
+
+    // XXX This should not be necessary, but ofp4 does not yet wait for OpenFlow flow table changes
+    // to commit before returning success.
+    std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
sounds like a bug, perhaps a new issue?

Reply:
Good suggestion, I filed https://github.com/vmware/nerpa/issues/86 and linked to it from the comment.